bukkit/shared/src/main/java/com/discordsrv/bukkit/listener/BukkitWorldChangeListener.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of DiscordSRV, licensed under the GPLv3 License
+ * Copyright (c) 2016-2025 Austin "Scarsz" Shapiro, Henri "Vankka" Schubin and DiscordSRV contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.discordsrv.bukkit.listener;
+
+import com.discordsrv.bukkit.BukkitDiscordSRV;
+import com.discordsrv.bukkit.debug.EventObserver;
+import com.discordsrv.common.abstraction.player.IPlayer;
+import com.discordsrv.common.core.logging.NamedLogger;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.player.PlayerChangedWorldEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class BukkitWorldChangeListener extends AbstractBukkitListener<PlayerChangedWorldEvent>{
+
+    public BukkitWorldChangeListener(BukkitDiscordSRV discordSRV) {
+        super(discordSRV, new NamedLogger(discordSRV, "WORLD_CHANGE_LISTENER"));
+    }
+
+    @EventHandler(priority = EventPriority.HIGH)
+    public void onPlayerQuit(PlayerChangedWorldEvent event) {
+        handleEventWithErrorHandling(event);
+    }
+
+    @Override
+    protected void handleEvent(@NotNull PlayerChangedWorldEvent event, Void __) {
+        IPlayer player = discordSRV.playerProvider().player(event.getPlayer());
+        discordSRV.eventBus().publish(new com.discordsrv.common.events.player.PlayerChangedWorldEvent(player));
+    }
+
+    private EventObserver<PlayerChangedWorldEvent, Boolean> observer;
+
+    @Override
+    protected void observeEvents(boolean enable) {
+        observer = observeEvent(observer, PlayerChangedWorldEvent.class, event -> event.getFrom() == null , enable);
```

Comment:
The cancellation condition is invalid. And observing cancellations of this event doesn't seem that useful, so this can be removed